File: frontend/src/api/agent.ts
import { API_CONFIG } from '@/config/api-config';
import { 
    Response, 
    AiAgentConfigResponseDTO, 
    CreateSessionResponseDTO, 
    ChatRequestDTO, 
    ChatResponseDTO 
} from '@/types/api';

// 统一处理 HTTP 层错误和后端业务错误，成功时保留完整 Envelope。
const handleResponse = async <T>(response: globalThis.Response):
    Promise<Response<T>> => {
    if (!response.ok) {
        const errorText = await response.text();
        throw new Error(`HTTP error! status: ${response.status}, message: ${errorText}`);
    }
    const data = await response.json();
    if (data.code !== "0000") {
        throw new Error(data.info || 'Unknown API error');
    }
    return data;
};

// 页面只依赖这个 API 对象，不直接关心 URL、fetch 和响应校验细节。
export const agentApi = {
    /**
     * Query AI Agent Config List
     * Path: /api/v1/query_ai_agent_config_list
     */
    queryAiAgentConfigList: async (): Promise<Response<AiAgentConfigResponseDTO[]>> => {
        const response = await fetch(`${API_CONFIG.BASE_URL}/query_ai_agent_config_list`, {
            method: 'GET',
            headers: {
                'Content-Type': 'application/json',
            },
        });
        return handleResponse<AiAgentConfigResponseDTO[]>(response);
    },

    /**
     * Create Session
     * Path: /api/v1/create_session
     */
    createSession: async (agentId: string, userId: string): Promise<Response<CreateSessionResponseDTO>> => {
        const response = await fetch(`${API_CONFIG.BASE_URL}/create_session`, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
            },
            body: JSON.stringify({ agentId, userId }),
        });
        return handleResponse<CreateSessionResponseDTO>(response);
    },

    /**
     * Chat
     * Path: /api/v1/chat
     */
    chat: async (data: ChatRequestDTO): Promise<Response<ChatResponseDTO>> => {
        const response = await fetch(`${API_CONFIG.BASE_URL}/chat`, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
            },
            body: JSON.stringify(data),
        });
        return handleResponse<ChatResponseDTO>(response);
    }
};

File: internal/api/dto/agent.go
package dto

// AiAgentConfigResponse 是查询 Agent 列表时返回给前端的精简视图。
type AiAgentConfigResponse struct {
	AgentID   string `json:"agentId"`
	AgentName string `json:"agentName"`
	AgentDesc string `json:"agentDesc"`
}

// CreateSessionRequest 描述创建会话接口需要的请求参数。
type CreateSessionRequest struct {
	AgentID string `json:"agentId"`
	UserID  string `json:"userId"`
}

// CreateSessionResponse 只向调用方返回新建或复用的 sessionId。
type CreateSessionResponse struct {
	SessionID string `json:"sessionId"`
}

// ChatRequest 描述同步聊天和流式聊天共用的请求结构。
type ChatRequest struct {
	AgentID   string `json:"agentId"`
	UserID    string `json:"userId"`
	SessionID string `json:"sessionId"`
	Message   string `json:"message"`
}

// ChatResponse 是非流式聊天接口的响应数据。
type ChatResponse struct {
	Content string `json:"content"`
}
